### DISCUSSION TITLE
Distances between labeled ground-truth and predicted labels

### DISCUSSION THREAD

**Discussion by ozgurcemozgen06** [talmolab/sleap#931] [General]
I have trained sleap on MARS dataset consisting of 15,000 two-mice images. However, as I inspect evaluation metric of "all distances" between predicted and ground-truth for validation set, I see that the shape of the array is sth like (2878,7). I have 7 body parts for each of the mice so, should not the second dimension of the array be 14? Also, I cannot understand the first dimension since my validation fraction was 0.1 so I believed it should have been 1,500. 

I have to reach the distances between labels of gt and predicted to compute precision and recall by comparing those distances with a decided threshold. So, is there any way to inspect through the labels.slp files?

**Comment by talmo** [comment]
Hi @ozgurcemozgen06,

The distances are `(n_imgs, n_nodes)` because we compute the Euclidean distance between the GT and predictions, so you get a scalar back for each point.

The 2878 might be the total number _instances_. Assuming there are some frames with just 1 animal, this works out as 1500 frames would have at the most 3000 instances. Alternatively, SLEAP filters out images with no animals before training, so that might also account for the discrepancy.

I'd also recommend checking out our notebook on [Model evaluation](https://sleap.ai/notebooks/Model_evaluation.html) as a reference for evaluation.

Let us know if you have any questions!
